diesel_codegen/src/schema_inference/pg.rs

```diff
@@ -38,11 +40,11 @@ table! {
 }
 
 pub fn determine_column_type(cx: &mut ExtCtxt, attr: &ColumnInformation) -> P<ast::Ty> {
-    let tpe = if attr.type_name.starts_with("_") {
-        let subtype = str_to_ident(&capitalize(&attr.type_name[1..]));
+    let tpe = if attr.type_name.ends_with("[]") {
```

Comment:
Returning the primary keys independently would be a way, but then bigger changes on sqlite side would be required. I will look into this tomorrow.